src/core/ComposableElement.php

```diff
@@ -293,6 +305,41 @@ private static function emptyInstance(): this {
     return $this->attributes->toMap();
   }
 
+  /**
+   * Determines if a given XHP attribute "key" represents an XHP spread operator
+   * in the constructor.
+   */
+  private static function isSpreadKey(string $key): bool {
+    return substr($key, 0, strlen(:xhp::SPREAD_PREFIX)) === :xhp::SPREAD_PREFIX;
```

Comment:
not worth doing another revision for this probably, but fwiw I wouldn't mind adding a dependency on the HSL to xhp-lib